index.html

```diff
@@ -1769,76 +1756,14 @@ <h4>
         <a data-link-for="NDEFRecord">recordType</a> attribute.
       </li>
-      <li>Switch on |target|:
-        <dl>
-          <dt>text</dt>
-          <ol>
-            <li>
-              If the |recordType| value is equal to "`empty`", return `null`.
-            </li>
-            <li>
-              If the |recordType| value is equal to "`text`", then run the following sub-steps:
-              <ol>
-                <li>
-                  Let |header| be the first <a>byte</a> of |bytes|.
-                </li>
-                <li>
-                  Let |charset| be "`utf-8`" if bit `7` (<a>MB field</a>) of
-                  |header| is equal to the value 0, or else "`utf-16be`".
-                </li>
-                <li>
-                  Let |offset| be the value given by bit `5` to bit `0` of the
-                  |header|.
-                </li>
-                <li>
-                  Let |buffer| be the |bytes|,
-                  from position |offset| + `1` to the end.
-                </li>
-                <li>
-                  If |charset| is equal to "`utf-8`", return the result of
-                  running <a>UTF-8 decode</a> on |buffer|.
-                </li>
-                <li>
-                  Otherwise, return the result of running <a>decode</a> on
-                  |buffer| with `encoding` set to "`utf-16be`".
-                </li>
-              </ol>
-            </li>
-            <li>
-              Otherwise, return the result of running <a>UTF-8 decode</a> on |bytes|.
-            </li>
-          </ol>
-          <dt>arrayBuffer</dt>
-          <ol>
-            <li>
-              Return an {{ArrayBuffer}} whose contents are the |bytes|.
-              Re-[= exception/throw =] any exceptions.
-            </li>
-          </ol>
-          <dt>JSON</dt>
-          <ol>
-            <li>
-              If the |recordType| value is equal to "`media`", then
-              return the result of running <a>parse JSON from bytes</a>
-              on |bytes|. Re-[= exception/throw =] any exceptions.
-            </li>
-            <li>
-              Otherwise, return `null`.
-            </li>
-          </ol>
-          <dt>records</dt>
-          <ol>
-            <li>
-              If the |recordType| value is "`smart-poster`", or "`media`", or
-              an <a>external type</a>, then return the result of running
-              <a>parse records from bytes</a> on |bytes|.
-              Re-[= exception/throw =] any exceptions.
-            </li>
-            <li>
-              Otherwise, return `null`.
-            </li>
-          </ol>
-        </dl>
+      <li>
+        If the |recordType| value is "`smart-poster`", or "`media`", or
+        an <a>external type</a>, then return the result of running
+        <a>parse records from bytes</a> on |bytes|.
+        Re-[= exception/throw =] any exceptions.
+      </li>
+      <li>
+        Otherwise, return `null`.
```

Comment:
I was told that throwing would be better. Just need to find the right exception

Reply:
[`OperationError`](https://heycam.github.io/webidl/#operationerror) or [`NotSupportedError`](https://heycam.github.io/webidl/#notsupportederror) are candidates.

Reply:
We are explicitly not supporting it (we could attempt on mime for instance) so better use NotSupportedError

Reply:
Let's go with `NotSupportedError`

Reply:
Fixed in https://github.com/w3c/web-nfc/pull/384/commits/58afbefc4b00a2554cd081c92bc01679f690b409